go/tasks/plugins/array/k8s/monitor.go

```diff
@@ -175,18 +188,27 @@ func LaunchAndCheckSubTasksState(ctx context.Context, tCtx core.TaskExecutionCon
 			}
 			return currentState, logLinks, subTaskIDs, err
 		}
-	}
 
-	newState = newState.SetArrayStatus(*newArrayStatus)
+		// validate map task parallelism
+		newSubtaskPhase := core.Phases[newArrayStatus.Detailed.GetItem(childIdx)]
+		if !newSubtaskPhase.IsTerminal() || newSubtaskPhase == core.PhaseRetryableFailure {
+			currentParallelism++
+		}
 
-	// Check that the taskTemplate is valid
-	taskTemplate, err := tCtx.TaskReader().Read(ctx)
-	if err != nil {
-		return currentState, logLinks, subTaskIDs, err
-	} else if taskTemplate == nil {
-		return currentState, logLinks, subTaskIDs, fmt.Errorf("required value not set, taskTemplate is nil")
+		if maxParallelism != 0 && currentParallelism >= maxParallelism {
+			// If max parallelism has been achieved we need to fill the subtask phase summary with
+			// the remaining subtasks so the overall map task phase can be accurately identified.
+			for i := childIdx + 1; i < len(currentState.GetArrayStatus().Detailed.GetItems()); i++ {
+				childSubtaskPhase := core.Phases[newArrayStatus.Detailed.GetItem(i)]
```

Comment:
it will happen in future rounds as you described. as implemented, this will just stop processing subtasks when we reach the max parallelism. however, without populating the summary array [calculating the map task phase](https://github.com/flyteorg/flyteplugins/blob/14ed6a8d2848fd200ac67f6bc35766ce48377b62/go/tasks/plugins/array/core/state.go#L256-L304) has unintended consequences. for example, failing because there are not enough tasks to achieve the minSuccesses.